ragna/ui/resources/upload.js

```diff
@@ -0,0 +1,51 @@
+
+
+function upload(files, token, informationsEndpoint, final_callback) {
+    var uploaded_documents = [] 
+
+    Array.from(files).map(file => {
+
+        getUploadInformations(file, token, informationsEndpoint).then(
+
+                function(uploadInformations) {
+                        uploadFile(uploadInformations.url, uploadInformations.data, file)
```

Comment:
Since the upload information includes the URL, we can safely remove the extra `uploadEndpoint` that we previously took as parameter to `upload`. For all non.default uploads that wouldn't have worked anyway, because we don't know the upload endpoint before retrieving the upload information.